File: C++/Easy/Accepted/check-if-one-string-swap-can-make-strings-equal_4.cpp
class Solution {
public:
    bool areAlmostEqual(string s1, string s2) {
        
        int n = s1.size();
        if (s1 == s2) return true;
        int cnt = 0;
        for (int i = 0; i < n; i++) {
            if (s1[i] != s2[i]) cnt++;
        }
        if (cnt == 2) {
            sort(s1.begin(), s1.end());
            sort(s2.begin(), s2.end());
            if (s1 == s2) return true;
        }
        return false;
    }
};

File: Java/Easy/Accepted/check-if-one-string-swap-can-make-strings-equal_4.java
class Solution {
    public boolean areAlmostEqual(String s1, String s2) {
        
        if (s1.equals(s2)) return true;
        if (s1.length() != s2.length()) return false;
        int diff = 0;
        char[] s1Chars = s1.toCharArray();
        char[] s2Chars = s2.toCharArray();
        for (int i = 0; i < s1Chars.length; i++) {
            if (s1Chars[i] != s2Chars[i]) {
                diff++;
            }
        }
        if (diff != 2) return false;
        Arrays.sort(s1Chars);
        Arrays.sort(s2Chars);
        return Arrays.equals(s1Chars, s2Chars);
    }
}